src/Classic.jl

```diff
@@ -38,63 +34,15 @@ abstract type Actor{TBehavior} end
 """
 abstract type Addr end
 
-abstract type SendStyle end
-struct Sendable <: SendStyle end
-struct CopySendable <: SendStyle end
-struct Racing <: SendStyle end
-struct NonSendable <: SendStyle end
-
-"""
-    Trait SendStyle: Sendable, CopySendable, Racing, NonSendable
-
-Trait for marking types that can be sent to actors.
-    
-Sending mutable data between actors breaks the model and it may lead to data races,
-so - as a general rule - it should be avoided.
-
-By default, every type is `NonSendable`. You have to mark yor message types as `Sendable`,
-`CopySendable` or `Racing` before sending them. e.g.
-
-```
-abstract type MyMessage end
-Classic.SendStyle(::Type{<:MyMessage}) = Racing()
-
-struct SafeMsg end
-Classic.SendStyle(::Type{SafeMsg}) = Sendable()
-```
-
-`CopySendable` types will be deep copied before sending.
-
-When using `Sendable`, you have to be sure that it is really safe to share the data. Note that
-immutable structs may also contain mutable data deeper down.
-
-With `Racing` you warn the user to trait the message as unsafe shared state. It will
-be delivered to a special `onmessage` method with an extra ::Racing argument for the
-receiver to be warned.
-"""
-SendStyle(::Type{<:Any}) = NonSendable()
-
 """
-    send(sender::Actor, target::Addr, msg)
+    send!(sender::Actor, target::Addr, msg)
```

Comment:
ok, I should have updated the docstring too. Agha gives as an example p.34
```
send content to customer
```
where `customer` is represented by its mailbox. Thus the mailbox is changed and the correct docstring would be
```
send!(recipient::Addr, msg)
```
If the sender wants to send its `Addr` to the recipient this has to go into the message. But he never will send its `Actor` variable. This must remain hidden, right?

BTW: This example shows that `Addr` is not a good naming, since an address it not changed but the mailbox. Therefore maybe `Mailbox` would be more appropriate. Agha calls that "link". Perhaps we should take `Link`.

Reply:
IMO the essence of asynchronous messaging is that sending a message does not change anything, just creates a new message that will be delivered somewhere in the future.

Yes, an address does not change when sending a message to it. Nor does the target mailbox or the target actor, because delivery to the mailbox will happen only in the future, possibly after the sending actor processed hundreds of other messages. Addititionally, the mailbox is not accessible from the sender, so even if it could change it the change itself would be not visible, so I think even in that case it would be meaningless to mark it as a changing operation.

(In Circo there is not even a notion of a mailbox)

BTW Agha calls the target "address". E.g. on p35: "The target must be a valid mail address. In other words, before an
actor can send the target a communication, it must know that the target
is a valid mail address [Hewitt and Baker 77]"  - I haven't yet read the original Hewitt paper, because Agha is the classic and I found another paper of Hewitt hard to read, but I will give it a try.

Reply:
Ok, I don't insist on both points. We can also go with common sense: we `send` mail to an address, not to a mailbox.